doc/sphinx/language/extensions/evars.rst

```diff
@@ -74,18 +74,51 @@ Expressions often contain redundant pieces of information. Subterms that can be
 automatically inferred by Coq can be replaced by the symbol ``_`` and Coq will
 guess the missing piece of information.
 
+Automatic resolution of existential variables
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Some tactics automatically resolve goals that are created when existential
+variables are created.
+
+.. _automatic-evar-resolution:
+
+.. example:: Automatic resolution of existential variables
+
+   .. coqtop:: reset in
+
+      Goal forall (a : nat), a = a -> exists n : nat, n = n.
+      Proof.
+        intros a e.
+
+   .. coqtop:: all
+
+       eexists ?[x].  (* creates a shelved goal *)
+       Unshelve.
+       exact e.  (* resolves the second goal automatically *)
+
+   Note the second goal would be resolved even without the :n:`Unshelve.`
+
+   The second goal (:n:`nat`) asks for a proof that :n:`nat` has a
+   :term:`witness`, i.e. it is not an empty type.  It can be proved directly
+   by applying a constructor of :n:`nat`, for example:
```

Comment:
Since you are pointing this out, you should also warn the user that solving this goal directly without carefully considering the other goals that depend on it is a typical case where the user can make a provable goal impossible to prove.

An illustration of this would be to add `3-4: exact 0.` after `Unshelve` in the example I gave above. This makes one of the remaining goals unprovable (0 < 0).

Reply:
Take a look at the example.

Reply:
Looks great!